terraform-bundle crashed

`terraform-bundle` on Terraform 0.10.1 when running within a golang:1.8.3 docker container.  There was no issue with 0.10.0.

*terrafor-bundle.hcl*
```
terraform {
  # Version of Terraform to include in the bundle. An exact version number
  # is required.
  version = "0.10.1"
}

providers {
  archive = ["~> 0.1"]
  aws = ["~> 0.1"]
  docker = ["~> 0.1"]
  dns = ["~> 0.1"]
  external = ["~> 0.1"]
  fastly = ["~> 0.1"]
  github = ["~> 0.1"]
  gitlab = ["~> 0.1"]
  grafana = ["~> 0.1"]
  http = ["~> 0.1"]
  ignition = ["~> 0.1"]
  kubernetes = ["~> 0.1"]
  local = ["~> 0.1"]
  mysql = ["~> 0.1"]
  pagerduty = ["~> 0.1"]
  random = ["~> 0.1"]
  rundeck = ["~> 0.1"]
  spotinst = ["~> 0.1"]
  template = ["~> 0.1"]
  terraform = ["~> 0.1"]
  tls = ["~> 0.1"]
}
```

```
$ docker run --rm -it -v `pwd`:/src --entrypoint /bin/sh golang:1.8.3
$ cd /src
$ go get github.com/hashicorp/terraform/tools/terraform-bundle
$ terraform-bundle package -os=linux -arch=amd64 terraform-bundle.hcl
Fetching Terraform 0.10.1 core package...
Fetching provider "docker"...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x9198ac]

goroutine 1 [running]:
github.com/hashicorp/terraform/plugin/discovery.(*ProviderInstaller).Get(0xc420693d88, 0xc42020a868, 0x6, 0xc42020c080, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/hashicorp/terraform/plugin/discovery/get.go:122 +0x4ec
main.(*PackageCommand).Run(0xc42020a440, 0xc4200100c0, 0x3, 0x3, 0x0)
	/go/src/github.com/hashicorp/terraform/tools/terraform-bundle/package.go:99 +0x90f
github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc420228240, 0xc420206db0, 0xc420191f68, 0xc420212498)
	/go/src/github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli/cli.go:242 +0x229
main.main()
	/go/src/github.com/hashicorp/terraform/tools/terraform-bundle/main.go:77 +0x2e0
```

--- a/tools/terraform-bundle/package.go
+++ b/tools/terraform-bundle/package.go
@@ -19,8 +19,6 @@ import (
 	"github.com/mitchellh/cli"
 )
 
-const releasesBaseURL = "https://releases.hashicorp.com"
-
 type PackageCommand struct {
 	ui cli.Ui
 }
@@ -94,16 +92,20 @@ func (c *PackageCommand) Run(args []string) int {
 		Ui:   c.ui,
 	}
 
+	if len(config.Providers) > 0 {
+		c.ui.Output(fmt.Sprintf("Checking for available provider plugins on %s...",
+			discovery.GetReleaseHost()))
+	}
+
 	for name, constraints := range config.Providers {
-		c.ui.Info(fmt.Sprintf("Fetching provider %q...", name))
 		for _, constraint := range constraints {
-			meta, err := installer.Get(name, constraint.MustParse())
+			c.ui.Output(fmt.Sprintf("- Resolving %q provider (%s)...",
+				name, constraint))
+			_, err := installer.Get(name, constraint.MustParse())
 			if err != nil {
-				c.ui.Error(fmt.Sprintf("Failed to resolve %s provider %s: %s", name, constraint, err))
+				c.ui.Error(fmt.Sprintf("- Failed to resolve %s provider %s: %s", name, constraint, err))
 				return 1
 			}
-
-			c.ui.Info(fmt.Sprintf("- %q resolved to %s", constraint, meta.Version))
 		}
 	}
 
@@ -184,7 +186,7 @@ func (c *PackageCommand) bundleFilename(version discovery.VersionStr, time time.
 func (c *PackageCommand) coreURL(version discovery.VersionStr, osName, archName string) string {
 	return fmt.Sprintf(
 		"%s/terraform/%s/terraform_%s_%s_%s.zip",
-		releasesBaseURL, version, version, osName, archName,
+		discovery.GetReleaseHost(), version, version, osName, archName,
 	)
 }
 

--- a/tools/terraform-bundle/package.go
+++ b/tools/terraform-bundle/package.go
@@ -91,6 +91,7 @@ func (c *PackageCommand) Run(args []string) int {
 
 		OS:   osName,
 		Arch: archName,
+		Ui:   c.ui,
 	}
 
 	for name, constraints := range config.Providers {

